src/environments/stubbed-environment-maker.js

```diff
@@ -0,0 +1,51 @@
+const fs = require('fs');
+const vm = require('vm');
+
+/**
+ * Creates a stubbed Sync Gateway environment for document definitions or a sync function for use in tests, schema
+ * validation, etc.
+ *
+ * @param {string} templateFile The absolute path to the template into which to insert the source contents
+ * @param {string} macroName The name of the macro to replace with the given source contents
+ * @param {string} sourceContents The raw string contents to be inserted into the template
+ * @param {string} [sourceFile] The optional path to the file from which the source contents originated, to be used to
+ *                              generate stack traces when errors occur
+ * @param {boolean} [unescapeBackticks] Whether backticks in the sync function string have been replaced with an escape
```

Comment:
This module should not have to be aware of any special handling for backtick characters. I will push that back to the `test-environment-maker` module in a subsequent commit.